### PR TITLE
refactor NSTemplateSet/NSTemplateTier Namespace type

### PR DESCRIPTION
**Pull request by xcoulon**
## Description
- rename 'Namespace' to 'NSTemplateSetNamespace' without
structural change
- duplicate 'NSTemplateSetNamespace' to 'NSTemplateSetTier'
and change the type of 'Template' from 'string' to
'templatev1.Template' (mandatory)

## Checks
1. Have you run `make generate` target? **yes**

2. Does `make generate` change anything in other projects (host-operator, member-operator)? **yes/**

3. In case other projects are changed, please provides PR links.
    - host-operator: https://github.com/codeready-toolchain/host-operator/pull/70
    - member-operator: N/A

## Notes
fixes CRT-273
Signed-off-by: Xavier Coulon <xcoulon@redhat.com>

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -13,6 +13,10 @@ require (
 	github.com/golang/protobuf v1.3.1 // indirect
 	github.com/google/gofuzz v1.0.0 // indirect
 	github.com/mailru/easyjson v0.0.0-20190403194419-1ea4449da983 // indirect
+	github.com/modern-go/reflect2 v1.0.1 // indirect
+	github.com/onsi/ginkgo v1.8.0 // indirect
+	github.com/onsi/gomega v1.5.0 // indirect
+	github.com/openshift/api v3.9.0+incompatible
```
Comment (MatousJobanek) [inline go.mod]: would be maybe good to alight the version with member operator: https://github.com/codeready-toolchain/member-operator/blob/master/go.mod#L14

File: pkg/apis/toolchain/v1alpha1/nstemplatetier_types.go [hunk 2]
```diff
@@ -13,7 +14,20 @@ type NSTemplateTierSpec struct {
...
+// NSTemplateTierNamespace the namespace definition in an NSTemplateTier resource
+type NSTemplateTierNamespace struct {
+
+	// The type of the namespace. For example: ide|cicd|stage|default
+	Type string `json:"type"`
+
+	// The revision of the corresponding template
+	Revision string `json:"revision"`
+
+	// Used to specify a custom template
```
[the remark this answers is not in the document]
Comment (xcoulon) [reply pkg/apis/toolchain/v1alpha1/nstemplatetier_types.go]: thanks @MatousJobanek ;)

File: go.mod (hunk 1 above)
Comment (MatousJobanek) [reply go.mod]: but it's maybe not a big deal

Comment (xcoulon) [reply go.mod]: ok, let me do that

Comment (xcoulon) [reply go.mod]: done in https://github.com/codeready-toolchain/api/commit/3d56a4420d7b77a216941e681abccda84a8312ef

**Pull request merged**